ci-operator/step-registry/telcov10n/sno-hub-ztp/set-hub-preserved/telcov10n-sno-hub-ztp-set-hub-preserved-commands.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ telcov10n cluster setup via agent command ************"
+# Fix user IDs in a container
+[ -e "${HOME}/fix_uid.sh" ] && "${HOME}/fix_uid.sh" || echo "${HOME}/fix_uid.sh was not found" >&2
+
+function setup_aux_host_ssh_access {
+
+  echo "************ telcov10n Setup AUX_HOST SSH access ************"
+
+  SSHOPTS=(
+    -o 'ConnectTimeout=5'
+    -o 'StrictHostKeyChecking=no'
+    -o 'UserKnownHostsFile=/dev/null'
+    -o 'ServerAliveInterval=90'
+    -o LogLevel=ERROR
+    -i "${CLUSTER_PROFILE_DIR}/ssh-key"
+  )
+
+}
+
+function set_expiration_time {
+
+  echo "************ telcov10n Set Hub cluster expiration time ************"
+
+  echo
+  set -x
+  b64_exp_time=$(echo "${EXPIRATION_TIME}" | base64 -w 0)
+  timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s --  \
+  "${NAMESPACE}" "${b64_exp_time}" "${EXPIRATION_TIME_FILE}" "${SHARED_HUB_CLUSTER_PROFILE}" << 'EOF'
+set -o nounset
+set -o errexit
+set -o pipefail
+
+exp_time_ns_file=/var/builds/${1}
+exp_time=$(echo ${2} | base64 -d)
+shared_exp_time_ns_file_sym_link=${3}
+cluster_profile_folder=$(dirname ${exp_time_ns_file})/${1}-cluster_profile_dir
+shared_cluster_profile_folder=${4}
+
+mkdir -pv $(dirname ${exp_time_ns_file})
+mkdir -pv ${cluster_profile_folder}
+
+TZ=UTC date --iso-8601=seconds -d "${exp_time}" > ${exp_time_ns_file}
+
+ln -sf ${exp_time_ns_file} ${shared_exp_time_ns_file_sym_link}
+ln -sf ${cluster_profile_folder} ${shared_cluster_profile_folder}
+
+echo "Now       : $(TZ=UTC date --iso-8601=seconds)"
+echo "Expire at : $(cat ${3}) [save at ${3} file]"
+EOF
+
+  set +x
+  echo
+}
+
+function save_cluster_profile_artifacts {
+
+  echo "************ telcov10n Save those artifacts that will be used during Spoke deployments ************"
+
+  hub_to_spoke_artifacts=(
+    "${KUBECONFIG}"
+    "${KUBEADMIN_PASSWORD_FILE}"
+    "${CLUSTER_PROFILE_DIR}"
+  )
+
+  echo
+  set -x
+  rsync -avP \
+    -e "ssh $(echo "${SSHOPTS[@]}")" \
+    "${hub_to_spoke_artifacts[@]}" \
+    "root@${AUX_HOST}":${SHARED_HUB_CLUSTER_PROFILE}/
```

Comment:
Can we use SHARED_HUB_CLUSTER_PROFILE only for the leave folder we should link so that we can hard-code `/var/telco-qe-preserved/` as the root where we want the symbolic links to be created?

Also.... I would like that the rsync happens to /var/builds/$NAMESPACE.

Reply:
Just one thing it don't understand properly yet. So, `/var/builds/$NAMESPACE` is a folder, not a file, am I right? I am a little bit confused about it. When I run this line above (being `${1}`, `${NAMESPACE}` actually):

```
exp_time_ns_file=/var/builds/${1}
. . .
TZ=UTC date --iso-8601=seconds -d "${exp_time}" > ${exp_time_ns_file}
```
I am trying to overwrite a folder with a file and that not make sense to me.

WRT cluster profile, the idea would be set this var like:

SHARED_HUB_CLUSTER_PROFILE: ztp-hub-preserved-prod-cluster_profile_dir

And then, from Spoke job, by defining the same var name, we will expect to get cluster profile info from:

`/var/telco-qe-preserved/${SHARED_HUB_CLUSTER_PROFILE}`

Is is ok?

Reply:
You should write to `/var/builds/$NAMESPACE/preserve`. `preserve` is the file that should exist to make the cluster preserved.

Reply:
Ok. Now the preserve part is clear.

What about the other question?

WRT cluster profile, the idea would be set this var like:

`SHARED_HUB_CLUSTER_PROFILE: ztp-hub-preserved-prod-cluster_profile_dir`

And then, from Spoke job, by defining the same var name, we will expect to get cluster profile info from:

`/var/telco-qe-preserved/${SHARED_HUB_CLUSTER_PROFILE}`

Is that ok?